File: src/main/java/com/ruby/pro203_exam/note/model/Note.java
package com.ruby.pro203_exam.note.model;

import com.ruby.pro203_exam.checker.model.PersonType;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "notes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Note {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "child_id")
    private UUID childId;  // NULL = kindergarten-wide note

    @Column(name = "kindergarten_id", nullable = false)
    private UUID kindergartenId;

    @Column(name = "title", length = 200)
    private String title;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(name = "note_date", nullable = false)
    private LocalDate noteDate;

    @Column(name = "created_by", nullable = false)
    private UUID createdBy;

    @Enumerated(EnumType.STRING)
    @Column(name = "created_by_type", nullable = false, length = 20)
    private PersonType createdByType;

    @Column(name = "created_at", nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/services/types/note.ts
export type PersonType = "Parent" | "Staff";

export interface CreateNoteDto {
    childId?: string;
    kindergartenId: string;
    title: string;
    content: string;
    noteDate: string;
}

export interface NoteResponseDto {
    id: string;
    childId?: string;
    childName?: string;
    kindergartenId: string;
    title: string;
    content: string;
    noteDate: string;
    createdBy: string;
    createdByType: PersonType;
    createdByName: string;
    createdAt: string;
    updatedAt?: string;
}